v20.1/alter-primary-key.md

```diff
@@ -0,0 +1,150 @@
+---
+title: ALTER PRIMARY KEY
+summary: Use the ALTER PRIMARY KEY statement to change the primary key of a table.
+toc: true
+---
+
+The `ALTER PRIMARY KEY` [statement](sql-statements.html) is a subcommand of [`ALTER TABLE`](alter-table.html) and can be used to change the [primary key](primary-key.html) of a table.
+
+
+
+{{site.data.alerts.callout_info}}
+To manage other constraints, see [`ADD CONSTRAINT`](add-constraint.html) and [`DROP CONSTRAINT`](drop-constraint.html).
+{{site.data.alerts.end}}
+
+## Synopsis
+
+<div>
+{% include {{ page.version.version }}/sql/diagrams/alter_primary_key.html %}
+</div>
+
+## Required privileges
+
+The user must have the `CREATE` [privilege](authorization.html#assign-privileges) on a table to alter its primary key.
+
+## Parameters
+
+| Parameter | Description |
+|-----------|-------------|
+| `table_name` | The name of the table with the primary key that you want to modify. |
+| `index_params` | The name of the column(s) that you want to use for the primary key. These columns replace the current primary key column(s). |
+| `opt_interleave` | You can potentially optimize query performance by [interleaving tables](interleave-in-parent.html), which changes how CockroachDB stores your data. |
+
+## Viewing schema changes
+
+{% include {{ page.version.version }}/misc/schema-change-view-job.md %}
+
+## Examples
+
+### Alter a single-column primary key
+
+Suppose that you are storing the data for users of your application in a table called `users`, defined by the following `CREATE TABLE` statement:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE TABLE users (
+  name STRING PRIMARY KEY,
+  email STRING
+);
+~~~
+
+The primary key of this table is on the `name` column. This is a poor choice, as some users likely have the same name, and all primary keys enforce a `UNIQUE` constraint on row values of the primary key column. Per our [best practices](performance-best-practices-overview.html#use-uuid-to-generate-unique-ids), you should instead use a `UUID` for single-column primary keys, and populate the rows of the table with generated, unique values.
+
+You can add a column and change the primary key with a couple `ALTER TABLE` statements:
+
+{% include copy-clipboard.html %}
+~~~ sql
+> ALTER TABLE users ADD COLUMN id UUID DEFAULT gen_random_uuid();
+~~~
+
+{% include copy-clipboard.html %}
+~~~ sql
+> ALTER TABLE users ALTER PRIMARY KEY USING COLUMN (id);
+~~~
+
+If you still plan to filter queries on the `name` column, you can add an index on the column to improve performance:
```

Comment:
unless you drop the old primary key (upcoming work), the primary key change is going to basically demote the old primary key into a secondary index, so there will already be an index on name post the primary key change.